### ISSUE TITLE
Plugin fails to generate changes when using rewrite-gradle

### ISSUE THREAD

**Issue by shanman190** [openrewrite/rewrite-gradle-plugin#83]
Given the sample:
https://github.com/shanman190/openrewrite-gradle-issue-gh-83

And executing `./gradlew rewriteDryRun` the plugin outputs:

```
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE
> Task :rewriteResolveDependencies

> Task :rewriteDryRun
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/<removed>/.gradle/caches/modules-2/files-2.1/org.openrewrite/rewrite-gradle/7.16.0/1a7a48ffbfdaeeb2ba79b1a0714f4e69d82b67e3/rewrite-gradle-7.16.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/<removed>/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-simple/1.7.30/e606eac955f55ecf1d8edcccba04eb8ac98088dd/slf4j-simple-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.gradle.internal.logging.slf4j.OutputEventListenerBackedLoggerContext]
Using active recipe(s) [com.example.VersionRelocation]
Using active styles(s) []
Validating active recipes
Running recipe(s)...
Applying recipes would make no changes. No report generated.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 24s
2 actionable tasks: 2 executed
```

As you can see in the sample, it should have rewritten the version from `2.3.12.RELEASE` to `2.4.13`. This same behavior also happens when trying to do `ChangeDependencyGroupId` and `ChangeDependencyArtifactId`. It feels like it may be that something with `org.openrewrite.gradle` effecting the `build.gradle` of itself is at play here?

**Comment by jkschneider** [comment]
Hi @shanman190. Can you try it with experimental Gradle parsing support on?

```groovy
rewrite {
  enableExperimentalGradleBuildScriptParsing = true
}
```

**Comment by shanman190** [comment]
So getting further, but this is entertaining.

```
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE
> Task :rewriteResolveDependencies

> Task :rewriteDryRun
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/tso4045/.gradle/caches/modules-2/files-2.1/org.openrewrite/rewrite-gradle/7.16.0/1a7a48ffbfdaeeb2ba79b1a0714f4e69d82b67e3/rewrite-gradle-7.16.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/tso4045/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-simple/1.7.30/e606eac955f55ecf1d8edcccba04eb8ac98088dd/slf4j-simple-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.gradle.internal.logging.slf4j.OutputEventListenerBackedLoggerContext]
Using active recipe(s) [com.example.VersionRelocation]
Using active styles(s) []
Validating active recipes
doesntmatter: 230: [Static type checking] - Cannot find matching method RewriteGradleProject#rewrite(groovy.lang.Closure). Please check if the declared type is correct and if the method exists.
 @ line 230, column 1.
   rewrite {
   ^

doesntmatter: 231: [Static type checking] - Cannot find matching method RewriteGradleProject#activeRecipe(java.lang.String). Please check if the declared type is correct and if the method exists.
 @ line 231, column 3.
     activeRecipe "com.example.VersionRelocation"
     ^

doesntmatter: 232: [Static type checking] - The variable [enableExperimentalGradleBuildScriptParsing] is undeclared.
 @ line 232, column 3.
     enableExperimentalGradleBuildScriptParsing = true
     ^

doesntmatter: 238: [Static type checking] - Cannot find matching method RewriteGradleProject#rewrite(java.lang.String). Please check if the declared type is correct and if the method exists.
 @ line 238, column 3.
     rewrite "org.openrewrite:rewrite-gradle:7.16.0"
     ^

4 errors

Running recipe(s)...
Applying recipes would make no changes. No report generated.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 20s
2 actionable tasks: 2 executed
```

1. Seems like extension parsing of itself is kind of unbalanced.
2. Should the build still be successful if there are errors? Even if it is just `rewriteDryRun`?

**Comment by shanman190** [comment]
So I did notice that I made a misstep when trying to use the `ChangeDependencyVersion` as it's parameter set is slightly different to that of `ChangeDependencyGroupId` and `ChangeDependencyArtifactId`. Correcting that error did in fact make the correct patch file for the version update. I'm going to re-review the group and artifact versions on my internal sample to see if that now works here too with the `enableExperimentalGradleBuildScriptParsing` turned on.
